controllers/storagecluster/storageclient.go

```diff
@@ -0,0 +1,63 @@
+package storagecluster
+
+import (
+	"fmt"
+
+	ocsclientv1a1 "github.com/red-hat-storage/ocs-client-operator/api/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/services"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	cutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	storageClientName = "local-storageclient"
+
+	tokenLifetimeInHours         = 48
+	onboardingPrivateKeyFilePath = "/etc/private-key/key"
+)
+
+type storageClient struct{}
+
+var _ resourceManager = &storageClient{}
+
+func (s *storageClient) ensureCreated(r *StorageClusterReconciler, storagecluster *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	if !storagecluster.Spec.AllowRemoteStorageConsumers {
+		r.Log.Info("Spec.AllowRemoteStorageConsumers is disabled")
+		return s.ensureDeleted(r, storagecluster)
+	}
+
+	storageClient := &ocsclientv1a1.StorageClient{}
+	storageClient.Name = storageClientName
+	storageClient.Namespace = r.OperatorNamespace
+	_, err := cutil.CreateOrUpdate(r.ctx, r.Client, storageClient, func() error {
+		if storageClient.Status.ConsumerID == "" {
```

Comment:
This is a bad indicator for starting the process of generating an onboarding token. You are assuming that immediately after setting an onboarding token the client gets reconciled and the status is updated. 
That is not the case, and there are not such guarantees. 

There are 2 options here:
1. check the spec 
2. have a combination of a UID + Status.phase check

Reply:
Thinking about this again, this check is not optimal but should work